core/src/main/java/org/mule/runtime/core/execution/ModuleFlowProcessingPhase.java

```diff
@@ -222,10 +231,22 @@ public void runPhase(final ModuleFlowProcessingPhaseTemplate template, final Mes
   }
 
   private void handleSourceError(final MessagingExceptionHandler exceptionHandler, Consumer<MessagingException> errorConsumer,
-                                 SourceErrorException see) {
-    MessagingException messagingException = see.toMessagingException();
-    exceptionHandler.handleException(messagingException, messagingException.getEvent());
-    errorConsumer.accept(messagingException);
+                                 MessagingException me,
+                                 Consumer<Either<MessagingException, Event>> terminateConsumer) {
+    SourceErrorException see = (SourceErrorException) me.getCause();
+    MessagingException exception = see.toMessagingException();
+
+    exceptionHandler.handleException(exception, exception.getEvent());
+    errorConsumer.accept(exception);
+    onTerminate(terminateConsumer, right(me.getEvent()));
+  }
+
+  private void handleErrorHandlingException(Consumer<MessagingException> errorConsumer,
+                                            Consumer<Either<MessagingException, Event>> terminateConsumer, MessagingException me,
+                                            PhaseResultNotifier phaseResultNotifier) {
+    phaseResultNotifier.phaseFailure((Exception) me.getCause());
+    SourceErrorException see = new SourceErrorException(me.getEvent(), sourceErrorResponseGenerateErrorType, me.getCause(), me);
+    onTerminate(terminateConsumer, left(see.toMessagingException()));
```

Comment:
This may be required, but it's not clear to me why this type of error should be handled via onTerminate, while other errors are handler via errorConsumer.  If there are things like parameter resosolution that shouldn't happen in the case of an error in error handler then shouldn't that be a condition (based on errorType) in error consumer?